incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +285,25 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
-func copyObject(inst *unstructured.Unstructured) *unstructured.Unstructured {
+func (r *ObjectReconciler) copyObject(nsNm string, inst *unstructured.Unstructured) *unstructured.Unstructured {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return nil
+	}
 	copied := inst.DeepCopy()
-
 	// Clear all irrelevant fields. cf https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#objectmeta-v1-meta
 	copied.SetCreationTimestamp(metav1.Time{})
-	copied.SetDeletionGracePeriodSeconds(nil)
-	copied.SetDeletionTimestamp(nil)
-	copied.SetFinalizers(nil) // TODO: double-check this is the right thing to do?
+	if copied.GetDeletionGracePeriodSeconds() != nil {
+		ns.SetCondition(forest.Local, tenancy.ObjectHasFinalizer, "finalizer set")
```

Comment:
For ```OBJECT_``` conditions, the key should be a string in the format of ```group/version/kind/namespace/name```. ```forest.Local``` is the key for locally-set namespace conditions, not object conditions.

You can find the documentation at #76

Reply:
I THINK I referenced this correctly.